force-app/main/default/triggers/MaintenanceRequest.trigger

```diff
@@ -0,0 +1,16 @@
+trigger MaintenanceRequest on Case (before update, after update) {
+
+    List<Case> validCaseList = new List<Case>();
+
+    if(Trigger.isUpdate && Trigger.isAfter) {
+        for(Case cases: Trigger.new){
+            if(cases.Status == 'Closed' && (cases.Type == 'Repair' || cases.Type == 'Routine Maintenance')){
+                validCaseList.add(cases);
+            }
+        }
+
+        if(!validCaseList.isEmpty()){
+            MaintenanceRequestHelper.createMaintenanceRequestRecord(validCaseList);
```

Comment:
it is not good practice to hold logic in Triggers, all what needs to be done in Trigger is call handler class where all the logic should be in pace, let's discuss also during the call